modules/ROOT/pages/migrate-client-apps.adoc

```diff
@@ -0,0 +1,154 @@
+* If the organization of the API you consume has registered an external client provider, for example, PingFederate or Okta, review the section: <<external_client,Organizations with Registered External Client Providers>>.
+* Do not attempt to update the client ID and client secret of an application.
++
+This feature is not supported.
+* Do not use the same user to create more than 500 applications.
++
+The user will not be able to create applications with the same name even if the client applications in the identity provider (IdP) with the same name and different client ID and client secret.
+* Do not attempt to refresh the client ID.
++
+This feature is not unless you delete and re-register the client application.
+
+[[migrate_with_internal]]
+== Migrate Client Applications for APIs with Native Client Providers
+
+To migrate your client application:
+
+. Open a text editor and paste the following commands:
++
+[source,text,linenums]
+----
+curl --location --request POST 'https://anypoint.mulesoft.com/exchange/api/v1/organizations/:orgId/applications' \
+--header 'Authorization: Bearer <omitted>' \
+--header 'Content-Type: application/json' \
+--data-raw '{
+    "clientId": <":myClientId>",
+    "clientSecret": ":myClientSecret",
+    "grantTypes": [],
+    "name": "My Application Name",
+    "redirectUri": []
+}'
+----
+. In the command snippet, replace the `orgId` parameter with your organization Id and the content of the body with the appropriate application <<properties,properties>>. 
++
+[NOTE]
+---
+You must use the organization ID of the master organization.
+---
+. Run this command at the Command prompt.
+
+[[properties]]
+== Properties Reference for the Migrate Task
+
+In the properties of the migrate command, provide values based on the following recommendations:
+
+* `clientId`: <Your client ID>
+* `clientSecret`: <Your client secret>
+* (optional)`description`: <Informative description for the applications>
++
+This property is displayed on the application page. 
+* `grantTypes`: As no IDP is provided, leave an empty array 
+* `name`: <Name of the application to be created>
++
+This property is displayed when selecting an application before requesting contracts.
+* `redirectUri`: <Array containing the redirect URIs for your application to use as an OAuth 2.0 client>
+* (optional) `url`: <URL of the client application>
++
+This property is displayed on the application page.
+
+[[migrate_with_external]]
+== Migrate Client Applications for APIs with External Client Providers
+
+When creating an application for an organization with registered external client providers, you must create the client application referencing the API that you want to consume. By doing so, you create the client application on the client provider to which the API is associated.
+
+When referencing an API that is associated to PingFederate external client provider or to the Anypoint Platform native client provider, you must include the client ID and client secret. 
```

Comment:
This is not correct. You are not forced too. As the docs mentions: `Including the client id/secret will only work when the referenced API is associated to PingFederate external client provider OR to Anypoint as the client provider.`

Reply:
Done! Thanks!